src/app/components/forms/Upload.tsx

```diff
@@ -1,31 +1,84 @@
+"use client";
 import Button from "@codegouvfr/react-dsfr/Button";
 import autoAnimate from "@formkit/auto-animate";
 import React, { InputHTMLAttributes, useEffect, useRef, useState } from "react";
 import Loader from "@/app/components/ui/Loader";
 import { cn } from "@/app/utils/classname.util";
+import { FileUploadResponse, useFileUpload } from "@/app/hooks/useFileUpload";
+import prettyBytes from "pretty-bytes";
 
 const Upload = ({
-  onFileChange,
-  onAbort,
   className,
   value,
+  accept,
+  onChange,
+  category,
   ...props
 }: UploadProps) => {
+  const { uploadFile, getFile, deleteFile } = useFileUpload();
+
+  const [state, setState] = useState<UploadStates>("idle");
+  const [errorMessage, setErrorMessage] = useState<string>("");
+  const [fileData, setFileData] = useState<FileDataType | undefined>(undefined);
```

Comment:
```suggestion
  const [fileData, setFileData] = useState<FileDataType | undefined>();
```
Puisque tu précises le type dans `useState`, normalement on peut un peu simplifier l'écriture (à tester)